qiskit_machine_learning/utils/loss_functions/loss_functions.py

```diff
@@ -13,46 +13,95 @@
 """ Loss utilities """
 
 from abc import ABC, abstractmethod
+from typing import Union, Tuple
 import numpy as np
 
+
 from ...exceptions import QiskitMachineLearningError
 
 
 class Loss(ABC):
     """
-    Abstract base class for Loss.
+    Abstract base class for computing Loss.
+
     """
 
-    def __call__(self, predict, target):
+    def __call__(self, predict: Union[int, np.ndarray], target: Union[int, np.ndarray]) -> float:
+        """
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Returns:
+            a float value of the loss function
+        """
         return self.evaluate(predict, target)
 
     @abstractmethod
-    def evaluate(self, predict, target):
-        """evaluate"""
+    def evaluate(self, predict: Union[int, np.ndarray], target: Union[int, np.ndarray]) -> float:
+        """
+        An abstract method for evaluating the loss function
+
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match
+        """
         raise NotImplementedError
 
     @abstractmethod
-    def gradient(self, predict, target):
-        """gradient"""
+    def gradient(self, predict: Union[int, np.ndarray], target: Union[int, np.ndarray]) -> float:
+        """
+        An abstract method for computing the gradient
+
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match.
+        """
         raise NotImplementedError
 
     @staticmethod
-    def _validate(predict, target):
+    def _validate(
+        predict: Union[int, np.ndarray], target: Union[int, np.ndarray]
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Returns:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match.
+        """
+
         predict = np.array(predict)
         target = np.array(target)
         if predict.shape != target.shape:
             raise QiskitMachineLearningError(
-                f"Shapes don't match, predict: {predict.shape}, " f"target: {target.shape}!"
+                f"Shapes don't match, predict: {predict.shape}, target: {target.shape}!"
             )
         return predict, target
 
 
 class L1Loss(Loss):
-    """L1Loss"""
+    """
+    L1Loss:
```

Comment:
The name of the class will be in the docs from the class itself, it does not need to be put into the docstring as it will not really add any value. All it needs is the description of what this class is about ie. the `L1Loss:` bit can simply be removed. Same in the other classes in this module.

Reply:
> 
> 
> The name of the class will be in the docs from the class itself, it does not need to be put into the docstring as it will not really add any value. All it needs is the description of what this class is about ie. the `L1Loss:` bit can simply be removed. Same in the other classes in this module.

Agree, docstrings are updated.